math/pythagoras.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2020 @author beqakd
+ * @file
+ *
+ * A basic implementation of 3d distance between two point.
+ */
+
+#include <cassert>   // For testing(assert)
+#include <cmath>     // For std::pow(), std::sqrt() and so on
+#include <iostream>  // For io operations
+
+/**
+ * The distance between two points in a three dimensional - 3D - coordinate
+ * system. For this we have struct point where we save coordinates of our points
+ * and later on calculate the distance as mentioned.
+ */
+class Distance {
+ public:
+    /**
+    * Point struct to save 3D coordinates.
+    */
+    struct Point {
```

Comment:
why a `struct` within a `class`?
`class` does not require this, just simply declare the member variables `x`, `y`,  and `z`